runtime/ms_rest_azure/lib/ms_rest_azure/resource.rb

```diff
@@ -2,14 +2,13 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 
+require_relative 'sub_resource'
+
 module MsRestAzure
   #
   # Class which represents any Azure resource.
   #
-  class Resource
-
-    # @return [String] the id of the resource.
-    attr_accessor :id
+  class Resource < SubResource

```

Comment:
@sarangan12 sorry, I'm not sure I understand your first comment here (in reply to Vishrut), can you clarify?
Also, what type of items are your referring to that need discussion for renaming?